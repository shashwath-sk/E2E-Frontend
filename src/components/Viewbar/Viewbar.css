.viewbar-container{
    background-color: rgb(39,39,39);
    height: 100vh;
    width: 300px;
}
.viewbar-header{
    height: 70px;
    background-color: rgb(80, 31, 198);
    font-size: 35px;
    color: white;
    padding-left: 20px;
    padding-top: 20px;
    margin-bottom: 40px;
}
.viewbar-contents{
    padding-left: 20px;
}
.viewbar-contents>*{
    color: rgb(90,90,90);
    font-size: 20px;
    padding-top:14px;
}
.viewbar-build-content{
    padding-left: 20px;
    color: rgb(90,90,90);
    margin-top: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}
.viewbar-content-focus{
    /* margin: 0px; */
    color: white;
    background-color:black
}
.viewbar-search{
    margin-left: 20px;
    padding-bottom: 10px;
    margin-right: 20px;
}
/* .view-bar-contents:hover{
    color: white;
    background-color:black
} */