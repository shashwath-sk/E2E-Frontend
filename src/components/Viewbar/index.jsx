import * as React from 'react';
import ContentTypeContext from '../../context/contentTypeContext';
import SearchField from '../../components/SearchField';
import './Viewbar.css';

export default function Viewbar() {
    const { ContentType } = React.useContext(ContentTypeContext);
    return(
        <div className='viewbar-container'>
            <div className='viewbar-header'>
                CMS+
            </div>
            <div className='viewbar-search'>
                <SearchField placeholder="COLLECTION TYPES" />
            </div>
            <div className='viewbar-contents'>
                {
                    ContentType && ContentType.map((content,index) => {
                        return (
                            <li key={index}>{content.Name}</li>
                        );
                    })
                }
            </div>
            <div className='viewbar-build-content viewbar-content-focus'>
                CONTENT TYPE BUILDER
            </div>
            
        </div>
    );

}