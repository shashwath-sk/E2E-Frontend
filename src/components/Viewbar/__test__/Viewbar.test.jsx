import Viewbar from '../';
import * as React from 'react';
import {render} from '@testing-library/react';
import { BrowserRouter} from 'react-router-dom';
// import makeRequest from '../../../utils/makeRequest';



describe('Viewbar', () => {
    it('should render Viewbar', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <Viewbar />
            </BrowserRouter>
        );
        expect(asFragment).toMatchSnapshot();
    });

    it('should show CMS+ as header when rendered', () => {
        const { getByText } = render(
            <BrowserRouter>
                <Viewbar />
            </BrowserRouter>
        );
        expect(getByText('CMS+')).toBeInTheDocument();
    });

    it('should show CONTENT TYPE BUILDER when rendered', () => {
        const { getByText } = render(
            <BrowserRouter>
                <Viewbar />
            </BrowserRouter>
        );
        expect(getByText('CONTENT TYPE BUILDER')).toBeInTheDocument();
    });

    it('should show search field with Collection types as placeholder when rendered', () => {
        const { getByPlaceholderText } = render(
            <BrowserRouter>
                <Viewbar />
            </BrowserRouter>
        );
        expect(getByPlaceholderText('COLLECTION TYPES')).toBeInTheDocument();
    }
    );




});