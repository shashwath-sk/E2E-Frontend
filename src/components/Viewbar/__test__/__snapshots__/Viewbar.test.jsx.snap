// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewbar should render Viewbar 1`] = `[Function]`;
